Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -335,11 +340,17 @@ bool MainWindow::Stop()
                                            "may be lost if you stop the current emulation "
                                            "before it completes. Force stop?") :
                                         tr("Do you want to stop the current emulation?"));
+
+    m_inquiry_in_progress = false;
+
     if (pause)
       Core::SetState(state);
 
     if (confirm != QMessageBox::Yes)
+    {
+      m_exit_requested = false;
```

Comment:
Okay, looking at this holistically, I think there's an issue with the fact that `Stop()`'s return value means both/either "we are not done stopping yet / The user doesn't want it stopped". What if we reorganized this logic to something like:

```c++
connect(this, &MainWindow::EmulationStopped, this, &MainWindow::DidStop);

bool MainWindow::DidStop() {
  m_stop_requested = false;
  m_render_widget->hide();
  if (m_exit_requested) close();
}

// This return value means "a stop was successfully requested". If true, DidStop() will
// be called at some point in the future.
bool MainWindow::RequestStop() {
  if (!Core::IsRunning()) {
    Core::QueueHostJob([this] { DidStop() }, true);
    return true;
  }

  // The rest of the current Stop() logic
}

bool MainWindow::eventFilter(QObject* object, QEvent* event)
{
  if (event->type() == QEvent::Close)
  {
    if (RequestStop() && object == this) m_exit_requested = true;
    static_cast<QCloseEvent*>(event)->ignore();
    return true;
  }
}
```

?